search-service/src/main/kotlin/com/egm/stellio/search/util/EntitiesQueryUtils.kt

```diff
@@ -216,29 +223,42 @@ fun buildTemporalQuery(
     val (timerel, timeAt) = buildTimerelAndTime(timerelParam, timeAtParam, inQueryEntities).getOrElse {
         return BadRequestDataException(it).left()
     }
+
+    if (timerel == Timerel.BETWEEN && endTimeAtParam == null)
+        return BadRequestDataException("'endTimeAt' request parameter is mandatory if 'timerel' is 'between'").left()
+
     if (withAggregatedValues && aggrMethodsParam == null)
         return BadRequestDataException("'aggrMethods' is mandatory if 'aggregatedValues' option is specified").left()
 
     val aggregate = aggrMethodsParam?.split(",")?.map {
-        if (TemporalQuery.Aggregate.isSupportedAggregate(it))
-            TemporalQuery.Aggregate.forMethod(it)!!
+        if (Aggregate.isSupportedAggregate(it))
+            Aggregate.forMethod(it)!!
         else
             return BadRequestDataException(
                 "'$it' is not a recognized aggregation method for 'aggrMethods' parameter"
             ).left()
     }
 
     val lastN = lastNParam?.toIntOrNull()?.let {
-        if (it >= 1) it else null
+        if (it > pagination.temporalLimitMax) return TooManyResultsException(
+            "You asked for the $it last temporal entities, but the supported maximum limit is ${
+                pagination.temporalLimitMax
+            }"
+        ).left()
+        else if (it >= 1) it
+        else null
     }
+    val limit = lastN ?: pagination.temporalLimitDefault
+    val asLastN = lastN != null
 
     return TemporalQuery(
         timerel = timerel,
         timeAt = timeAt,
         endTimeAt = endTimeAt,
         aggrPeriodDuration = aggrPeriodDurationParam,
         aggrMethods = aggregate,
-        lastN = lastN,
+        limit = limit,
```

Comment:
maybe `temporalLimit` would be a better name (to not confuse with the `limit` query parameter which acts on the number of returned entities)?

Reply:
it will do temporalQuery.temporalLimit maybe temporalQuery.instanceLimit? not sure